test/Fail/Issue1967.err

```diff
@@ -5,7 +5,9 @@ type, are never considered by instance search, so having an
 instance argument ⦃ A : Set ⦄ has no effect.
 when checking that the expression ⦃ A : Set ⦄ → Set is a type
 
-Issue1967.agda:5.22-23: error: [ShouldBePi]
-Set should be a function type, but it isn't
+Issue1967.agda:5.22-23: error: [CannotApply]
+Expression used as function but does not have function type:
```

Comment:
I apologize for the bikeshedding, but how about wording this as "[expression] does not take an argument" or "[expression] is not a function"? Also perhaps we could add "(did you supply too many arguments?)" like GHC does.

Reply:
> Also perhaps we could add "(did you supply too many arguments?)" like GHC does.

I am not so sure this is so helpful for Agda, because we do not have static arities.  E.g. the arity can increase when you refine an argument.  So the error may not be that there are too many arguments, but that there is some mistake in a previous argument blocking reduction of the type to a function type.

In general, I am not sure Agda should give hints without having done some analysis that the hint goes into the right direction.  Hints leading in the wrong direction are annoying, so one is better off without a hint.

Reply:
> how about wording this as "[expression] does not take an argument" or "[expression] is not a function"?

The introductory text _"Expression used as function but does not have function type:"_ has to be seen in connection to the data supplied in the following: expression and type.  This sentence sets the stage for these two actors appearing after it.
I am open to reformulations but they should rework the whole message.

Also, I think we should get away from terms embedded in prose and rather separate prose and Agda terms.  This new error is an attempt of doing so.

Reply:
I am experimenting with this.  Maybe the `ReallyNotBlocked` criterium is still too naive.  E.g. we get the hint in this situation:
```agda
postulate
  D : {{A : Set}} → Set
  test : (A : Set) → D A
```
The problem is though not that `D` is applied to too many arguments (as the hint says), but it is applied with the _wrong hiding_.